python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -1029,11 +1035,36 @@ def _evaluate_auto_materialize_tick(
                 for rr in run_requests
             ]
 
+            print_group_name = self._get_print_sensor_name(sensor)
+
+            self._logger.info(
+                "Tick produced"
+                f" {len(run_requests)} run{'s' if len(run_requests) != 1 else ''} and"
+                f" {len(evaluations_by_key)} asset"
+                f" evaluation{'s' if len(evaluations_by_key) != 1 else ''} for evaluation ID"
+                f" {evaluation_id}{print_group_name}"
+            )
+
+            # Fetch all data that requires the code server before writing the cursor, to minimize
+            # the chances that changes to code servers after the cursor is written (e.g. a
+            # code server moving into an error state or an asset being renamed) causes problems
+            workspace = workspace_process_context.create_request_context()
+            for run_request in run_requests:
+                if not run_request.requires_backfill_daemon():
+                    get_job_execution_data_from_run_request(
```

Comment:
The other potential downside here is that if the code server calls are flaky/timing out and failing some % of the time (which I don't think is at all common, but maybe still worth considering) - before it would have made some progress before proceeding until the first call failed, now it would need every call to succeed before proceeding